docs/nextflow/bundle-scripts.md

```diff
@@ -0,0 +1,84 @@
+
+## Script location and behavior
+
+Nextflow pipelines can store `bin` scripts in multiple locations. Wave treats each location differently:
+
+| Location                                                   | Nextflow project path             | Wave behavior         |
+|------------------------------------------------------------|-----------------------------------|-----------------------|
+| [Project `bin` directory](#project-bin-directory)          | `${projectDir}/bin/`              | Bundled conditionally |
+| [Module `bin` directory](#module-bin-directory)            | `${moduleDir}/resources/usr/bin/` | Always bundled        |
+
+### Project `bin` directory
+
+Wave does not bundle scripts from the project [`bin` directory](https://nextflow.io/docs/latest/sharing.html#the-bin-directory) by default. Nextflow uploads these scripts to the work directory at runtime using cloud APIs. This approach adds network overhead and can be inefficient when launching many tasks.
+
+When you enable Fusion or use the AWS Fargate executor, Wave bundles scripts from the project `bin` directory into a container layer. This provides better performance but ties scripts to specific execution environments.
+
+Modifying scripts in the project `bin` directory modify the container fingerprint for all Wave containers in the workflow. See [Wave container fingerprinting](#wave-container-fingerprinting) for more information.
+
+### Module `bin` directory
+
+Wave bundles scripts from [module `bin` directories](https://nextflow.io/docs/latest/module.html#module-binaries) into container layers. Module `bin` directories are a portable script solution that works across execution environments.
+
+You must enable Wave and the module binaries feature to bundle module scripts:
+
+```groovy
+nextflow.enable.moduleBinaries = true
+
+wave {
+  enabled = true
+}
+```
+Scripts are scoped to specific modules and only affect containers that use those modules. Modifying a script in a module `bin` directory only changes the container fingerprint for processes that use that module, leaving other containers unchanged. See [Wave container fingerprinting](#wave-container-fingerprinting) for more information.
+
+:::warning
+Module binaries do not work on cloud executors without Wave.
+:::
+
+## Wave container fingerprinting
+
+Wave bundles `bin` scripts into container layers and generates fingerprints that become part of the task hash:
+
+- Scripts in module `bin` directory are added as a container layer
+- When Fusion or the AWS Fargate executor are enabled, scripts in project `bin` directory are added as a container layer
+- Script layers receive a fingerprint based on content, ignoring file timestamps
+- Script layer fingerprints are incorporated in the container fingerprint
+
+If you modify `bin` scripts, Wave generates a new layer fingerprint, which creates a new container fingerprint and invalidates the cache.
```

Comment:
I assume this is the Wave cache? Link to materials that describe how the Wave cache works?